src/LitePatch/LitePatch.Services/Services/GitPatchService.cs

```diff
@@ -1,38 +1,43 @@
 ﻿using System.Management.Automation;
+using LibGit2Sharp;
 using LitePatch.Services.Interfaces;
+using LitePatch.Services.Models;
+using LitePatch.Services.Repo;
 
 namespace LitePatch.Services.Services;
 
-public class GitPatchService(IGitInfoService gitInfoService, ISettingsService settingsService) : IGitPatchService
+public class GitPatchService() : IGitPatchService
 {
-    private int _counter = 1;
-    
-    public void CreatePatch(string sha, string commitName)
-    {
-        var output = Path.Combine(settingsService.Settings.OutputFolderPath,
-            $"{_counter.ToString().PadLeft(4, '0')}-{CleanPatchName(commitName)}.patch");
-        
-        var command = @$"git format-patch -1 -B1% {sha} --output='{output}'";
 
-        using var powershell = PowerShell.Create();
-        powershell.AddScript(@$"cd {gitInfoService.RepositoryPath}");
-        powershell.AddScript(command);
+    private readonly IPatchRepository _patchRepository;
+    private readonly ISettingsService _settingsService;
 
-        var results = powershell.Invoke();
-        _counter++;
+    public GitPatchService(IPatchRepository patchRepository, ISettingsService settingsService) : this()
+    {
+        _patchRepository = patchRepository;
+        var outputFolderPath = settingsService.Settings.OutputFolderPath;
```

Comment:
I was thinking that too. It could look something like this: 

![image](https://github.com/d3molite/LitePatch/assets/63230618/160b679c-02d2-4d16-9c84-6c440855e243)